command/secrets/init/secrets_init.go

```diff
@@ -1,25 +1,71 @@
 package init
 
 import (
+	"fmt"
+
 	"github.com/0xPolygon/polygon-edge/command"
 	"github.com/spf13/cobra"
 )
 
+const (
+	// maxInitNum is the maximum value for "num" flag
+	maxInitNum = 1000
+)
+
+var (
+	errInvalidNum = fmt.Errorf("num flag value should be between 1 and %d", maxInitNum)
+)
+
 func GetCommand() *cobra.Command {
+	var (
+		basicParams initParams
+		number      int
+	)
+
 	secretsInitCmd := &cobra.Command{
 		Use: "init",
 		Short: "Initializes private keys for the Polygon Edge (Validator + Networking) " +
 			"to the specified Secrets Manager",
-		PreRunE: runPreRun,
-		Run:     runCommand,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Why not extract this into the separate method:
This is the standard we've been using so far and i think its useful
ibftSnapshotCmd := &cobra.Command{
		Use:     "propose",
		Short:   "Proposes a new candidate to be added or removed from the validator set",
		PreRunE: runPreRun,
		Run:     runCommand,
	}

Reply:
Done: https://github.com/0xPolygon/polygon-edge/pull/770/commits/df3173e8ff57e8e2ef4bc4f66b8b4f077087a325